lib/diff.ml

```diff
@@ -110,6 +135,16 @@ and module_declaration ~typing_env ~name ~reference ~current =
       modtype_item ~loc:reference.md_loc ~typing_env ~name
         ~reference:ref_modtype ~current:curr_modtype
 
+and module_type_declaration ~typing_env ~name ~reference ~current =
+  match (reference.mtd_type, current.mtd_type) with
+  | Some (Mty_signature ref_sub), Some (Mty_signature curr_sub) ->
+      signatures ~typing_env ~reference:ref_sub ~current:curr_sub
+      |> Option.map (fun mdiff -> Module { mname = name; mdiff })
+  | Some ref_sub, Some curr_sub ->
+      modtype_item ~loc:reference.mtd_loc ~typing_env ~name ~reference:ref_sub
+        ~current:curr_sub
+  | _ -> assert false
```

Comment:
The `mtd_type` field is an option because a module type can be abstract, just like a type:

```
(* in an interface file *)
module type S

module M : S
```

So the `None` cases are possible and you need to handle them!

Reply:
> ```
> module type P = sig val x : int end  
>
> becomes 
>
> module type P
> ```

Yes, exactly. This is an API change, so you should return `Some`!

Reply:
If that gets too complicated, you could also simply compare concrete module types and leave the handling of `abstract <-> concrete` diffs for a later PR. If you go down that road, I think it would still be good to add a test that shows that this change is not handled!